Hazel/src/Hazel/Events/KeyEvent.h

```diff
@@ -1,34 +1,35 @@
 #pragma once
 
 #include "Event.h"
+#include "Hazel/Input.h"
 
 namespace Hazel {
 
 	class HAZEL_API KeyEvent : public Event
 	{
 	public:
-		inline int GetKeyCode() const { return m_KeyCode; }
+		inline KeyCode GetKeyCode() const { return m_KeyCode; }
 
 		EVENT_CLASS_CATEGORY(EventCategoryKeyboard | EventCategoryInput)
 	protected:
-		KeyEvent(int keycode)
+		KeyEvent(KeyCode keycode)
 			: m_KeyCode(keycode) {}
 
-		int m_KeyCode;
+		KeyCode m_KeyCode;
 	};
 
 	class HAZEL_API KeyPressedEvent : public KeyEvent
 	{
 	public:
-		KeyPressedEvent(int keycode, int repeatCount)
+		KeyPressedEvent(KeyCode keycode, int repeatCount)
 			: KeyEvent(keycode), m_RepeatCount(repeatCount) {}
 
 		inline int GetRepeatCount() const { return m_RepeatCount; }
 
 		std::string ToString() const override
 		{
 			std::stringstream ss;
-			ss << "KeyPressedEvent: " << m_KeyCode << " (" << m_RepeatCount << " repeats)";
+			ss << "KeyPressedEvent: " << static_cast<uint32_t>(m_KeyCode) << " (" << m_RepeatCount << " repeats)";
```

Comment:
Instead of this it might be worth overloading operator<<(std::ostream&,KeyCode) for the enum

Reply:
Now that the overload operator is implemented (in https://github.com/TheCherno/Hazel/pull/124/commits/e10bfd761b2b43d6540236e13c0e38f0515260ae), you don't have to static cast anymore when using the `<<` operator, so you can remove those casts as well.